src/fmu/dataio/providers/_fmu.py

```diff
@@ -248,26 +250,36 @@ def _get_restart_data_uuid(self) -> UUID | None:
         """Load restart_from information"""
         assert self._runpath is not None
         logger.debug("Detected a restart run from environment variable")
-        restart_path = self._runpath / os.environ[RESTART_PATH_ENVNAME]
-        restart_case_metafile = (
-            restart_path.parent.parent / ERT_RELATIVE_CASE_METADATA_FILE
-        ).resolve()
+        restart_path = (self._runpath / os.environ[RESTART_PATH_ENVNAME]).resolve()
 
-        if not restart_case_metafile.exists():
+        if _casepath_has_metadata(restart_path.parent.parent):
+            restart_case_metafile = (
+                restart_path.parent.parent / ERT_RELATIVE_CASE_METADATA_FILE
+            )
+        else:
             warn(
-                f"{RESTART_PATH_ENVNAME} environment variable is set to "
-                f"{os.environ[RESTART_PATH_ENVNAME]} which is invalid. Metadata "
-                "restart_from will remain empty.",
+                f"Environment variable {RESTART_PATH_ENVNAME} resolves to the path "
+                f"{restart_path} which is non existing or points to an ERT run "
+                "without case metdata. Metadata 'restart_from' will remain empty.",
                 UserWarning,
             )
             return None
 
-        restart_metadata = schema.InternalCaseMetadata.model_validate(
-            ut.yaml_load(restart_case_metafile, loader="standard")
-        )
-        return _utils.uuid_from_string(
-            f"{restart_metadata.fmu.case.uuid}{restart_path.name}"
-        )
+        try:
```

Comment:
wrapped this in `try/except` ATM to avoid failures due to this.. I foresee that a restart ensemble might get a bit old on disk and potentially have invalid case metadata. The only thing we are going to pick up is the `case uuid`, probably we don't need to parse it with the full `InternalCaseMetadata` maybe it is better to `yaml_load` it and use `get()` syntax to fetch the uuid. Will create separate issue on it